src/main/java/org/openrewrite/java/testing/junit5/AddParameterizedTestAnnotation.java

```diff
@@ -0,0 +1,116 @@
+    private static final AnnotationMatcher ENUM_SOURCE_MATCHER = new AnnotationMatcher(PARAMS_PATH+"EnumSource");
+    private static final AnnotationMatcher CSV_FILE_SOURCE_MATCHER = new AnnotationMatcher(PARAMS_PATH+"CsvFileSource");
+    private static final AnnotationMatcher ARG_SOURCE_MATCHER = new AnnotationMatcher(PARAMS_PATH+"ArgumentsSource");
+    private static final AnnotationMatcher PARAM_TEST_MATCHER = new AnnotationMatcher("@org.junit.jupiter.params.ParameterizedTest");
+
+    @Override
+    public String getDisplayName() {
+        return "Add missing @ParameterizedTest annotation when @ValueSource is used or " +
+               "replace @Test with @ParameterizedTest";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Add missing @ParameterizedTest annotation when @ValueSource is used or " +
+               "replace @Test with @ParameterizedTest.";
+    }
+
+    @Override
+    public JavaIsoVisitor<ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            // checks if annotations matches ValueSource and siblings
+            public boolean checkForSourceAnnotation(J.Annotation ann) {
+                // must be a better way to writer this
+                return (VALUE_SOURCE_MATCHER.matches(ann) || CSV_SOURCE_MATCHER.matches(ann) ||
+                        METHOD_SOURCE_MATCHER.matches(ann) || NULL_SOURCE_MATCHER.matches(ann) ||
+                        EMPTY_SOURCE_MATCHER.matches(ann) || NULL_EMPTY_SOURCE_MATCHER.matches(ann) ||
+                        ENUM_SOURCE_MATCHER.matches(ann) || CSV_FILE_SOURCE_MATCHER.matches(ann) ||
+                        ARG_SOURCE_MATCHER.matches(ann));
+            }
+
+            public List<J.Annotation> reorderAnnotations(List<J.Annotation> annotations) {
+                // if @ParameterizedTest is already at start then return early
+                if (PARAM_TEST_MATCHER.matches(annotations.get(0))) { return annotations; }
+                List<J.Annotation> ordered = new ArrayList<>(annotations);
+                // get @ParameterizedTest
+                List<J.Annotation> filteredList = ordered.stream().filter(PARAM_TEST_MATCHER::matches).collect(Collectors.toList());
+                J.Annotation paramAnn = filteredList.get(0);
+
+                ordered.remove(paramAnn);
+
+                return ordered;
+            }
+
+            @Override
+            public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration md, ExecutionContext ctx) {
+                J.MethodDeclaration m = super.visitMethodDeclaration(md, ctx);
+
+                // return early if @ValueSource and siblings are not detected
+                if (m.getLeadingAnnotations().stream().noneMatch(this::checkForSourceAnnotation) ||
+                    m.getLeadingAnnotations().stream().anyMatch(PARAM_TEST_MATCHER::matches)) {
```

Comment:
I'd flip these two around, as the first one is more likely and easier to test, so probably faster.
```suggestion
                if (m.getLeadingAnnotations().stream().anyMatch(PARAM_TEST_MATCHER::matches) ||
                    m.getLeadingAnnotations().stream().noneMatch(this::checkForSourceAnnotation)) {
```